witness/src/Dscp/DB/Rocks/Real/Functions.hs

```diff
@@ -46,28 +77,64 @@ closeNodeDB = closeRocksDB . _rdDatabase
 -- Reading/writing
 ------------------------------------------------------------
 
--- | Read ByteString from RocksDb using given key.
-rocksGetBytes :: MonadIO m => ByteString -> DB -> m (Maybe ByteString)
-rocksGetBytes key DB {..} = Rocks.get rocksDB rocksReadOpts key
+get :: (HasRocksDB ctx m, Serialise a, Serialise b) => a -> m (Maybe b)
+get a = do
+    db  <- getDB
+    mbs <- rocksGetBytes db (serialise' a)
+    case mbs of
+        Just it -> case deserialiseOrFail (LBS.fromStrict it) of
+            Left  err -> throwM (DeserialisationError err)
+            Right res -> return (Just res)
 
--- | Write ByteString to RocksDB for given key.
-rocksPutBytes :: MonadIO m => ByteString -> ByteString -> DB -> m ()
-rocksPutBytes k v DB {..} = Rocks.put rocksDB rocksWriteOpts k v
+        Nothing -> do
+            return Nothing
 
--- | Delete element from RocksDB for given key.
-rocksDelete :: MonadIO m => ByteString -> DB -> m ()
-rocksDelete k DB {..} = Rocks.delete rocksDB rocksWriteOpts k
+put :: (HasRocksDB ctx m, Serialise a, Serialise b) => [(a, b)] -> m ()
+put pairs = do
+    db <- getDB
 
-------------------------------------------------------------
--- Instances
-------------------------------------------------------------
+    rocksWriteBatch db $ map (Rocks.Del . serialise') pairs
 
-getDB :: (MonadReader ctx m, HasLens' ctx RocksDB) => m DB
-getDB = view $ lensOf @RocksDB . rdDatabase
+-- TODO(kirill.andreev): Put proper mass-erase here.
+delete :: (HasRocksDB ctx m, Serialise a) => [a] -> m ()
+delete keyz = do
```

Comment:
It's not like it is really important, but why did you name it `keyz`? Is it supposed to be a joke?

Reply:
Universum exports `keys` function :peka:

Reply:
We'll need a really big ~flamethrower~ team to refactor that.